google-cloud-bigtable/lib/google/cloud/bigtable/encryption_info.rb

```diff
@@ -0,0 +1,118 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require "google/cloud/bigtable/status"
+
+module Google
+  module Cloud
+    module Bigtable
+      ##
+      # # EncryptionInfo
+      #
+      # Encryption information for a given resource.
+      #
+      # See {Backup#encryption_info} and {Table::ClusterState#encryption_infos}.
+      #
+      # @example
+      #   require "google/cloud/bigtable"
+      #
+      #   bigtable = Google::Cloud::Bigtable.new
+      #   instance = bigtable.instance("my-instance")
+      #   cluster = instance.cluster("my-cluster")
+      #
+      #   backup = cluster.backup("my-backup")
+      #
+      #   encryption_info = backup.encryption_info
+      #   encryption_info.encryption_type #=> :GOOGLE_DEFAULT_ENCRYPTION
+      #
+      # @example Retrieve a table with cluster states containing encryption info.
+      #   require "google/cloud/bigtable"
+      #
+      #   bigtable = Google::Cloud::Bigtable.new
+      #
+      #   table = bigtable.table("my-instance", "my-table", view: :ENCRYPTION_VIEW, perform_lookup: true)
+      #
+      #   table.cluster_states.each do |cs|
+      #     puts cs.cluster_name
+      #     puts cs.encryption_infos.first.encryption_type
+      #   end
+      #
+      class EncryptionInfo
+        # @private
+        #
+        # Creates a new EncryptionInfo instance.
+        def initialize grpc
+          @grpc = grpc
+        end
+
+        ##
+        # The type of encryption used to protect the resource. Possible values:
+        #
+        # * `ENCRYPTION_TYPE_UNSPECIFIED` - Encryption type was not specified, though data at rest remains encrypted.
+        # * `GOOGLE_DEFAULT_ENCRYPTION` - The data backing the resource is encrypted at rest with a key that is
+        #   fully managed by Google. No key version or status will be populated. This is the default state.
+        # * `CUSTOMER_MANAGED_ENCRYPTION` - The data backing the resource is encrypted at rest with a key that is
+        #   managed by the customer. The in-use version of the key and its status are populated for CMEK-protected
+        #   tables. CMEK-protected backups are pinned to the key version that was in use at the time the backup was
+        #   taken. This key version is populated but its status is not tracked and is reported as `UNKNOWN`.
+        #
+        # See also {#encryption_status}, {#kms_key_version} and {Instance::ClusterMap#add}.
+        #
+        # @return [Symbol] The encryption type code as an uppercase symbol.
+        #
+        def encryption_type
```

Comment:
forgive my lack of ruby knowledge - are these wrappers necessary? Can we call directly on the generated types in this case?

Reply:
The generated types are completely encapsulated in all of the handwritten Ruby client libraries.